### PR TITLE
[ember__routing] update Transition API

### PR DESCRIPTION
**Pull request by BarryThePenguin**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://api.emberjs.com/ember/3.21/classes/Transition
https://github.com/tildeio/router.js/blob/master/lib/router/transition.ts
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/ember__routing/-private/transition.d.ts [hunk 1]
```diff
@@ -1,12 +1,26 @@
 import RouteInfo from './route-info';
 
-export default interface Transition {
+export default interface Transition<T = any> extends Partial<Promise<T>> {
```
Comment (BarryThePenguin) [inline types/ember__routing/-private/transition.d.ts:3]: I took this from

https://github.com/tildeio/router.js/blob/bb16c56396ef68416fac80fe7e4fb493ce2bed42/lib/router/transition.ts#L42

Which also exports a `type PublicTransition = Transition<any>;`

https://github.com/tildeio/router.js/blob/bb16c56396ef68416fac80fe7e4fb493ce2bed42/lib/router/transition.ts#L20

I left the type arguments in, as they may be useful in the future?

File: types/ember__routing/tsconfig.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
         "module": "commonjs",
         "target": "es5",
         "lib": [
-            "es6",
+            "es2018",
```
Comment (BarryThePenguin) [inline types/ember__routing/tsconfig.json:6]: `Promise.prototype.finally` is only available with `es2018`, I'm not sure what else this impacts though 💭

**Review by jamescdavis — CHANGES_REQUESTED** [review]
Some small test improvements, please. 😃

File: types/ember__routing/-private/transition.d.ts [hunk 3]
```diff
@@ -16,11 +30,85 @@ export default interface Transition {
...
+        label?: string,
+    ): Promise<TResult1 | TResult2>;
+    /**
+     * Fires an event on the current list of resolved/resolving handlers within this transition.
+     * Useful for firing events on route hierarchies that haven't fully been entered yet.
+     *
+     * @param  ignoreFailure a boolean specifying whether unhandled events throw an error
+     * @param  name          the name of the event to fire
+     */
+    send(ignoreFailure: boolean, name: string): void;
```
[the remark this answers is not in the document]
Comment (BarryThePenguin) [reply types/ember__routing/-private/transition.d.ts:102]: Sure thing! Done 👍

**Review by jamescdavis — APPROVED** [review]
Thank you for this!

**Comment by BarryThePenguin** [comment]
Ready to merge

**Review by amcasey — APPROVED** [review]

**Comment by BarryThePenguin** [comment]
Ready to merge

**Pull request merged**